bin/analyze.py

```diff
@@ -99,10 +107,22 @@ def main():
             masses = analysis.load.load_masses(cg, topology=traj.top)
         print('Loaded masses')
 
-        # keep only the lipids
-        sel_atoms = traj.top.select("(not name water) and " +
+        if selection_string == 'all_lipids':
+            # keep only the lipids
+            sel_atoms = traj.top.select("(not name water) and " +
                                         "(not resname tip3p " +
                                         "HOH SOL)")
+        else:
+            print('Parsing trajectory based on selection: ', selection_string)
+            sel_atoms = traj.top.select("( " + selection_string + ")")
+        
+        # very simple check to ensure that the selection actually
+        # contains some atoms atom_slice will fail if the array
+        # size is 0, this will at least provide some more specific feedback
+        # as to why
+        if len(sel_atoms) == 0:
+            print("Error: selection does not include any atoms.")
```

Comment:
This should raise an Exception. 
```suggestion
            raise ValueError("Error: selection does not include any atoms.")
```